File: java/statePattern/SystemsGo.java
package statePattern;

public class SystemsGo implements State {
	private MissionControl missionControl;

	public SystemsGo(MissionControl missionControl) {
		this.missionControl = missionControl;
	}

	@Override
	public void checkAllSystems() {
		System.out.println("All systems are ready.");
		this.missionControl.setState(this.missionControl.getStartPowerUnits());
	}

	@Override
	public void startPowerUnits() {
		System.out.println("Systems are not ready yet");

	}

	@Override
	public void retractArms() {
		System.out.println("Systems are not ready yet");

	}

	@Override
	public void shutDown() {
		System.out.println("All systems are to cease operations immediatly.");
		this.missionControl.setState(this.missionControl.getAbortProcedure());
	}

	@Override
	public void rocketBooster() {
		System.out.println("Systems are not ready yet");

	}

}

File: php/statePattern/SystemsGo.php
<?php  namespace php\statePattern;

class SystemsGo implements State {
	
	public $missionControl;

	public function __construct(MissionControl $missionControl) {
		$this->missionControl = $missionControl;
	}

	public function checkAllSystems() {
		echo "All system are ready<br>";
		$this->missionControl->setState($this->missionControl->getStartPowerUnits());
	}
	
	public function startPowerUnits() {
		echo "systems are not ready yet";
	}
	
	public function retractArms() {
		echo "systems are not ready yet";
	}
	
	public function shutDown() {
		echo "All systems are to cease operations immediatly.<br>";
		$this->missionControl->liftOffHandOver($this->missionControl->getAbortProcedure());
	}
	
	public function rocketBooster() {
		echo "systems are not ready yet";
	}
}
